src/Functions/IFunctionCustomWeek.h

```diff
@@ -116,6 +127,11 @@ class IFunctionCustomWeek : public IFunction
                 getName(), arguments.size());
     }
 
+    virtual std::pair<Field, Field> convertStringToUnixTimestamp(const Field & left, const Field & right) const
```

Comment:
Why does this need to be a virtual function? Are there any customization points in derived classes that might want to change this behavior?

Reply:
@Enmk Yes, only `toDayOfWeek` should deal with string arguments, other derived class for function like `toStartOfWeek` do not accept String as its argument

Reply:
@Enmk Good question!  I found that I've made it more complicated. I'll  rewrite this.